ext/drreg/drreg.c

```diff
@@ -2009,8 +2009,13 @@ drreg_exit(void)
             return DRREG_ERROR;
     }
 
-    /* Support re-attach */
-    memset(&ops, 0, sizeof(ops));
+    /* Support re-attach. */
+    if (dr_doing_detach()) {
+        memset(&ops, 0, sizeof(ops));
+#ifdef DEBUG
+        stats_max_slot = 0;
+#endif
```

Comment:
(OK it does seem like the other ones are set at init time:
    122:static drreg_options_t ops;
    124:static int tls_idx = -1;
    125:static uint tls_slot_offs;
    126:static reg_id_t tls_seg;
    129:static uint stats_max_slot;
   1843:static int drreg_init_count;
   1845:static per_thread_t init_pt;
)

Reply:
Yep.